src/torrent.cpp

```diff
@@ -2717,6 +2723,20 @@ namespace {
 			valid_endpoints++;
 		});
 
+#if TORRENT_USE_RTC
+		if(auto pos = url.find(':'); pos != string_view::npos)
+		{
+			string_view const protocol(url.data(), pos);
+			if (protocol == "ws" || protocol == "wss")
+			{
+				// WebSocket trackers will ignore the endpoint anyway
+				valid_endpoints = std::min(valid_endpoints, std::size_t(1));
```

Comment:
Unlike the standard HTTP and UDP trackers, Websocket trackers do not care about endpoints, since webRTC connections choose it themselves automatically with ICE. This change is to do only one WebSocket request provided there is a valid endpoint.

This could be replaced by leaving the tracker request loop after the first endpoint, would it be better?

Reply:
these are *local* endpoints. It's probably a bit misleading name, it's more like "listen sockets". Normal trackers announce once from each eligible local socket/endpoint. e.g. if you have one IPv4 and one IPv6 IP address; when you announce to a tracker, you need to announce using both as source address. It generalizes to the case where you may have multiple IP addresses of any version though.

Supporting this may not be very high priority for webtorrents, but I would imagine that in principle you'd like to have one tracker connection per local listen socket. To emulate having incoming connections connect to you on every listen interface/IP address you have.

Reply:
Actually that's the point, webtorrent trackers are different in this regard. You only need one tracker connection irrelevant of the number of local endpoints, and you never advertise your IP addresses to the tracker. The reason is webRTC offer/answer (which contain the candidate IP addresses) and the following connectivity checks will choose the proper local endpoint to use.

Reply:
that would be a great comment in the code!